SPECS/selinux-policy/0038-enable-liveos-iso-flow.patch

```diff
@@ -0,0 +1,459 @@
+From 5d7d3ac384cf2549b62ee5466bfbf6ef12b82f75 Mon Sep 17 00:00:00 2001
+From: George Mileka <gmileka@microsoft.com>
+Date: Fri, 22 Nov 2024 17:04:06 -0800
+Subject: [PATCH] Enable SELinux for the LiveOS ISO scenario.
+
+In the LiveOS ISO scenario, the initrd is configured to run several systemd daemons,
+then pivots to the rootfs where the same daemons are re-started. When SELinux is
+enabled, the daemons fail on the second run because some of the targets (mainly
+file descriptors) are already tagged kernel_t and there is no policy to allow those
+daemons to access those targets.
+
+This fix allows this flow - which in turn allows the LiveOS ISO scenario with SELinux
+enabled.
+---
+ policy/modules/admin/cloudinit.te    |  1 +
+ policy/modules/admin/rpm.te          |  1 +
+ policy/modules/admin/usermanage.te   |  2 ++
+ policy/modules/apps/loadkeys.te      |  2 ++
+ policy/modules/kernel/devices.if     | 18 ++++++++++++++++++
+ policy/modules/kernel/kernel.if      | 19 +++++++++++++++++++
+ policy/modules/kernel/kernel.te      |  1 +
+ policy/modules/services/chronyd.te   |  1 +
+ policy/modules/services/cron.te      |  1 +
+ policy/modules/services/dbus.te      |  1 +
+ policy/modules/services/ssh.te       |  2 ++
+ policy/modules/system/authlogin.te   |  1 +
+ policy/modules/system/fstools.te     |  1 +
+ policy/modules/system/getty.te       |  2 ++
+ policy/modules/system/hostname.te    |  1 +
+ policy/modules/system/locallogin.te  |  1 +
+ policy/modules/system/logging.te     |  2 ++
+ policy/modules/system/mount.te       |  1 +
+ policy/modules/system/selinuxutil.te |  2 ++
+ policy/modules/system/systemd.te     | 14 ++++++++++++++
+ policy/modules/system/udev.te        |  2 ++
+ 21 files changed, 76 insertions(+)
+
+diff --git a/policy/modules/admin/cloudinit.te b/policy/modules/admin/cloudinit.te
+index 94f7eabd2..2529970f2 100644
+--- a/policy/modules/admin/cloudinit.te
++++ b/policy/modules/admin/cloudinit.te
+@@ -50,6 +50,7 @@ dontaudit cloud_init_t self:capability { net_admin sys_admin sys_tty_config };
+ allow cloud_init_t self:fifo_file rw_fifo_file_perms;
+ allow cloud_init_t self:unix_dgram_socket create_socket_perms;
+ allow cloud_init_t self:passwd passwd;
++allow cloud_init_t kernel_t:fd use;
```

Comment:
In `init.te`, there are two type attributes (sets) that can be used to apply this to all the relevant domains: `daemon` and `systemprocess`. You can add `kernel_use_fds(daemon)` and `kernel_use_fds(systeprocess)` to blanket these permissions appropriately.

Reply:
pending

Reply:
fixed.